### ISSUE TITLE
Rule Change: [no-useless-computed-key] warn about double square brackets

### ISSUE THREAD

**Issue by k-yle** [eslint/eslint#17536]
### What rule do you want to change?

`no-useless-computed-key`

### What change to do you want to make?

Generate more warnings

### How do you think the change should be implemented?

A new default behavior

### Example code

```js
x = { [['abc']]: 123 } // -> x = { 'abc': 123 }

x = { [[[[[[[['abc']]]]]]]]: 123 } // -> x = { 'abc': 123 }
```

### What does the rule currently do for this code?

nothing

### What will the rule do after it's changed?

remove all nested arrays

### Participation

- [X] I am willing to submit a pull request to implement this change.

### Additional comments

One potential problem is that this cannot be safely autofixed if `Array#toString` has been overriden.

e.g.
```js
Array.prototype.toString = () => "A"

x = { [['B']]: 1 } // -> { A: 1 }
```

**Comment by mdjermanovic** [comment]
Per our rules policies, stylistic rules are [frozen](https://github.com/eslint/eslint#stylistic-rule-updates) so we're not making any changes in this rule except for bug fixes and adding support for new ECMAScript features. I don't think this could be considered a bug because this rule suggests removing `[]` of the computed keys syntax only where it can be done without changing the code between the `[]`, while this would also include removing `[]` of arrays. You can always copy the core rule into a custom rule and make it behave the way you want.
